src/UnitTests/Core/CLanguage/CParserTests.cs

```diff
@@ -1136,5 +1136,18 @@ public void CParser_Regression()
             var sExp = "(decl Char ((init-decl (ptr (func get ((Int n)))))))";
             Assert.AreEqual(sExp, decl.ToString());
         }
+
+        [Test]
+        public void CParser_IncorrectStatement()
+        {
+            Lex("typedef a b;");
+            Decl decl;
+            try {
+                decl = parser.Parse_ExternalDecl();
+            } catch (Exception) {
+                decl = null;
+            }
+            Assert.IsNull(decl);
+        }
```

Comment:
I' m not sure that it should be illegal. I just want to fix for infinite loop. So you could fix it as you think to be correct.